### ISSUE TITLE
Error compiling Compiling Swift Module 'OpenSSL' 

### ISSUE THREAD

**Issue by siuying** [Zewo/Examples#3]
I get error trying to run build with ``swift build``. I'm on OSX and install openssl with brew. Any tips?

```
Compiling Swift Module 'OpenSSL' (9 sources)
/Users/siuying/workspace/web/Examples/Packages/COpenSSL-0.1.0/module.modulemap:2:12: error: header '/usr/include/openssl/evp.h' not found
    header "/usr/include/openssl/evp.h"
           ^
/Users/siuying/workspace/web/Examples/Packages/OpenSSL-0.1.2/Sources/OpenSSL.swift:32:8: error: could not build Objective-C module 'COpenSSL'
import COpenSSL
       ^
/Users/siuying/workspace/web/Examples/Packages/COpenSSL-0.1.0/module.modulemap:2:12: error: header '/usr/include/openssl/evp.h' not found
    header "/usr/include/openssl/evp.h"
           ^
/Users/siuying/workspace/web/Examples/Packages/OpenSSL-0.1.2/Sources/OpenSSL.swift:32:8: error: could not build Objective-C module 'COpenSSL'
import COpenSSL
       ^
/Users/siuying/workspace/web/Examples/Packages/COpenSSL-0.1.0/module.modulemap:2:12: error: header '/usr/include/openssl/evp.h' not found
    header "/usr/include/openssl/evp.h"
           ^
/Users/siuying/workspace/web/Examples/Packages/OpenSSL-0.1.2/Sources/OpenSSL.swift:32:8: error: could not build Objective-C module 'COpenSSL'
import COpenSSL
       ^
/Users/siuying/workspace/web/Examples/Packages/COpenSSL-0.1.0/module.modulemap:2:12: error: header '/usr/include/openssl/evp.h' not found
    header "/usr/include/openssl/evp.h"
           ^
/Users/siuying/workspace/web/Examples/Packages/OpenSSL-0.1.2/Sources/OpenSSL.swift:32:8: error: could not build Objective-C module 'COpenSSL'
import COpenSSL
       ^
/Users/siuying/workspace/web/Examples/Packages/COpenSSL-0.1.0/module.modulemap:2:12: error: header '/usr/include/openssl/evp.h' not found
    header "/usr/include/openssl/evp.h"
           ^
/Users/siuying/workspace/web/Examples/Packages/OpenSSL-0.1.2/Sources/OpenSSL.swift:32:8: error: could not build Objective-C module 'COpenSSL'
import COpenSSL
       ^
/Users/siuying/workspace/web/Examples/Packages/COpenSSL-0.1.0/module.modulemap:2:12: error: header '/usr/include/openssl/evp.h' not found
    header "/usr/include/openssl/evp.h"
           ^
/Users/siuying/workspace/web/Examples/Packages/OpenSSL-0.1.2/Sources/OpenSSL.swift:32:8: error: could not build Objective-C module 'COpenSSL'
import COpenSSL
       ^
/Users/siuying/workspace/web/Examples/Packages/COpenSSL-0.1.0/module.modulemap:2:12: error: header '/usr/include/openssl/evp.h' not found
    header "/usr/include/openssl/evp.h"
           ^
/Users/siuying/workspace/web/Examples/Packages/OpenSSL-0.1.2/Sources/OpenSSL.swift:32:8: error: could not build Objective-C module 'COpenSSL'
import COpenSSL
       ^
/Users/siuying/workspace/web/Examples/Packages/COpenSSL-0.1.0/module.modulemap:2:12: error: header '/usr/include/openssl/evp.h' not found
    header "/usr/include/openssl/evp.h"
           ^
/Users/siuying/workspace/web/Examples/Packages/OpenSSL-0.1.2/Sources/OpenSSL.swift:32:8: error: could not build Objective-C module 'COpenSSL'
import COpenSSL
       ^
<unknown>:0: error: build had 1 command failures
swift-build: exit(1): ["/Library/Developer/Toolchains/swift-2.2-SNAPSHOT-2015-12-01-a.xctoolchain/usr/bin/swift-build-tool", "-f", "/Users/siuying/workspace/web/Examples/.build/debug/OpenSSL.o/llbuild.yaml"]
```

**Comment by paulofaria** [comment]
For now we're only supporting OpenSSL on Linux. @alex-alex Have you tried it on OSX?

**Comment by alex-alex** [comment]
`brew` installs openssl in `/usr/local/Cellar/openssl/` where Swift can't find

You'll have to copy/link headers to `/usr/include/openssl` and libssl with libcrypto to `/usr/local/lib`

**Comment by siuying** [comment]
I see. Since I'm using 10.11, I cant copy file to ``/usr/include``.

I just wonder why other library (http_parser, venice) works while openssl not. Can we have the modulemap file point to a different path only on OSX?

**Comment by paulofaria** [comment]
Use `sudo` to copy the files. The other libraries work because we own them, so we can choose where to install them. Obviously we install them in the same location on OSX and Linux. Currently it's not possible to deal with different paths on different OSs. One way would be to create a separate module with the correct path for OSX. But I'm not sure, just changing the path would solve the issue. @alex-alex Would it?

**Comment by siuying** [comment]
@paulofaria after 10.11 you cant copy files there even sudo, need to run [something in Recovery Mode](https://www.quora.com/How-do-I-turn-off-the-rootless-in-OS-X-El-Capitan-10-11)

it seems swift-package-manager do not have a solution to this problem [yet](https://github.com/apple/swift-package-manager/blob/master/Documentation/SystemModules.md#cross-platform-module-maps)

**Comment by paulofaria** [comment]
Maybe you should try a different Package with a different modulemap then. But I'm not sure.

**Comment by ermalkaleci** [comment]
Build succeed on OS X 10.10 after:
```bash
brew install openssl postgresql && brew link openssl --force
```

**Comment by alex-alex** [comment]
Yes, this was fixed by Zewo/OpenSSL@701314c

**Comment by siuying** [comment]
Good to know, thanks！
